src/bootstrap/lib.rs

```diff
@@ -1336,18 +1336,30 @@ impl Build {
     }
 
     fn beta_prerelease_version(&self) -> u32 {
+        fn extract_beta_rev<P: AsRef<Path>>(version_file: P) -> Option<String> {
+            let version = fs::read_to_string(version_file).ok()?;
```

Comment:
If this function takes the file content directly rather than taking file path and reading it later, we can write unit tests on this with prodiving dummy file content as an function argument.

Reply:
Done. And also added tests.